### ISSUE TITLE
[no-output-native] False positive when EventEmitter is of an object type with specific variable names

### ISSUE THREAD

**Issue by coder925** [angular-eslint/angular-eslint#544]
**Description and reproduction of the issue**
Even if my property is named `myOutput`, the linter says it is named as standard DOM event.
However, the type definition of my _EventEmitter_ has property names `start` and `end` which causes the error.

***Repro steps***
1. `ng new MyApp`
2. `ng add @angular-eslint/schematics`
3. Add to _app.component.ts_:
```TS
@Output() myOutput: EventEmitter<{start: Date, end: Date}> = new EventEmitter<{start: Date, end: Date}>();
```

***Expected result:***
- No errors since the variable is not named as a standard DOM event.

***Actual result:***
```
  11:3  error  Outputs should not be named or aliased as standard DOM event  @angular-eslint/no-output-native

✖ 1 problem (1 error, 0 warnings)

Lint errors found in the listed files.
```
***Workaround 1:***
Rename object type variable names:
`@Output() myOutput: EventEmitter<{starter: Date, ender: Date}> = new EventEmitter<{starter: Date, ender: Date}>();`

***Workaround 2:***
1. Extract object type into an interface file dateRange.ts:
```
export interface DateRange {
    start: Date;
    end: Date;
}
```
2. Change code to use interface:
```TS
@Output() myOutput: EventEmitter<DateRange> = new EventEmitter<DateRange>();
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `4.23.0` |
| `@typescript-eslint/parser`        | `4.23.0` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `14.15.3` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 12.0.4
Node: 14.15.3
Package Manager: npm 6.14.9
OS: win32 x64

Angular: 12.0.4
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.4
@angular-devkit/build-angular   12.0.4
@angular-devkit/core            12.0.4
@angular-devkit/schematics      12.0.4
@schematics/angular             12.0.4
rxjs                            6.6.7
typescript                      4.2.4

```

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by coder925** [comment]
Related: #543 
(created just before my issue)

**Comment by rafaelss95** [comment]
I'm closing this is as it's a duplicate of #524 and #543.
